client/src/components/board.rs

```diff
@@ -70,7 +68,7 @@ fn get_dragged_piece_position(
     let center = get_center(
         &to_position(cx, mouse_down),
         cx.props.size,
-        cx.props.perspective,
+        **use_shared_state::<Perspective>(cx).unwrap().read(),
```

Comment:
Haha this is cool, but do we know it doesn't cause issues calling here? Also maybe we could make a helper/macro for this.